superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -1318,32 +1318,36 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
                   validationErrors={validationErrors}
                 />
                 <div css={(theme: SupersetTheme) => infoTooltip(theme)}>
-                  <Button
-                    data-test="sqla-connect-btn"
-                    buttonStyle="link"
-                    onClick={() =>
-                      setDB({
-                        type: ActionType.configMethodChange,
-                        payload: {
-                          engine: db.engine,
-                          configuration_method:
-                            CONFIGURATION_METHOD.SQLALCHEMY_URI,
-                          database_name: db.database_name,
-                        },
-                      })
-                    }
-                    css={buttonLinkStyles}
-                  >
-                    {t(
-                      'Connect this database with a SQLAlchemy URI string instead',
-                    )}
-                  </Button>
-                  <InfoTooltip
-                    tooltip={t(
-                      'Click this link to switch to an alternate form that allows you to input the SQLAlchemy URL for this database manually.',
-                    )}
-                    viewBox="0 -6 24 24"
-                  />
+                  {dbModel.name !== 'Google Sheets' && (
```

Comment:
can we check the engine against `gsheets` instead?
It's done like that in the same file, so if we can, we should use the same checking method.
And probably move that `gsheets` engine string to a constant.